### ISSUE TITLE
Ensure kitchen sink works with PHP 7.2 and 7.3 for CWP2

### ISSUE THREAD

**Issue by chillu** [silverstripe/cwp-recipe-kitchen-sink#22]
### Overview

_Updated description 19-03-19_

PHP 7.0 is no longer supported, and PHP 7.1 is going EOL in Dec 2019. Ideally we're testing the oldest and newest supported PHP 7.x release in our modules, so at the moment PHP 7.1 and PHP 7.3 - but have a look at PHP 7.2 as well because it's technically supported.

Note that we need to keep testing 4.x on PHP 5.6 at least until we officially remove support: https://www.silverstripe.org/blog/our-plan-for-ending-php-5-6-support-in-silverstripe-4/

We need to discover compatibility issues on our supported modules early (e.g. mcrypt extension removal). Core supports PHP 7.2 already, meaning we should run builds here.

### Notes
- CWP1 builds captured here: https://github.com/silverstripe/cwp-recipe-kitchen-sink/issues/31
- Core modules are out of scope - they're tracked separately here: https://github.com/silverstripe/silverstripe-framework/issues/8864

It looks like RealMe is the only module preventing this issue from closing?
- Related: https://github.com/silverstripe/silverstripe-realme/issues/24
- https://github.com/silverstripe/silverstripe-realme/issues/17

Related issue for framework and open source modules: https://github.com/silverstripe/silverstripe-framework/issues/9116

**Comment by ScopeyNZ** [comment]
I know one module that does not yet have 7.2 compatibility: https://github.com/silverstripe/silverstripe-realme/issues/24

**Comment by robbieaverill** [comment]
We may as well add 7.3 too while we're at it

**Comment by stojg** [comment]
I assume as more and more sites are moving to 7.1 we will also see more of reports like this: 
![image](https://user-images.githubusercontent.com/485130/49966953-66312f00-ff86-11e8-895a-0f4e728af95b.png)

**Comment by robbieaverill** [comment]
@stojg FYI we've updated hybridsessions so it doesn't use mcrypt by default any more

**Comment by robbieaverill** [comment]
I've updated this to include 7.3

**Comment by robbieaverill** [comment]
Test run with 7.2 and 7.3 in it: https://travis-ci.org/creative-commoners/cwp-recipe-kitchen-sink/builds/486844029

This would be blocked by https://github.com/silverstripe/silverstripe-realme/issues/24 at least

**Comment by chillu** [comment]
Could you try running a build with everything *apart* from the realme module? We need to identify how much work is ahead of us, rather than just fixing one blocker to discover the next blocker ;)

**Comment by robbieaverill** [comment]
Yep

**Comment by chillu** [comment]
Alright, so looks like from a unit test perspective, RealMe is the *only* blocker for PHP 7.3 support across our supported modules? There might be stuff that's not covered by unit tests of course.

Do you know if we've ever actively looked for deprecated PHP features across supported modules?
https://secure.php.net/manual/en/migration71.deprecated.php
https://secure.php.net/manual/en/migration72.deprecated.php
https://secure.php.net/manual/en/migration73.deprecated.php

**Comment by robbieaverill** [comment]
I think we did for PHP 7.1 (e.g. with replacing mcrypt in hybridsessions) but I'm not sure we've done that for PHP 7.2 or 7.3 yet

**Comment by sminnee** [comment]
Have fixed the impact. Given that PHP 7.1 EOLs in 9 months and people often that long to upgrade, this is very important.

**Comment by sminnee** [comment]
Have split out CWP1 to #33

**Comment by chillu** [comment]
I've split out core module travis builds into https://github.com/silverstripe/silverstripe-framework/issues/8864

**Comment by brynwhyman** [comment]
We'll take this issue for supported module builds for CWP2 then.

**Comment by brynwhyman** [comment]
This is still blocked by the RealMe work. Testing hasn't gone as smoothly as we would have hoped, but we'll be picking this up in the next week or so and seeing it through.

**Comment by brynwhyman** [comment]
Missed this issue. This work was completed in other issues/ PRs.
